### PR TITLE
Add pagination, sorting and swagger

### PR DESCRIPTION
**Pull request by VArabchuk**

### CODE REVIEW

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 1]
```diff
@@ -18,42 +21,50 @@
...
 @RequiredArgsConstructor
 @RequestMapping("/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (sarakhmen) [inline src/main/java/mate/academy/intro/controller/BookController.java]: please add default pageable values

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 2]
```diff
@@ -18,42 +21,50 @@
...
             @PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto
     ) {
         return bookService.updateById(id, requestDto);
     }
 
     @GetMapping("/search")
+    @Operation(summary
+            = "Search for books", description = "Searches for books based on specific parameters")
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
         return bookService.search(searchParameters);
```
Comment (sarakhmen) [inline src/main/java/mate/academy/intro/controller/BookController.java]: what about pagination here?

**Review by sarakhmen — CHANGES_REQUESTED** [review]
I don't see swagger configuration within this PR. Could you please expose it here?

File: src/main/java/mate/academy/intro/controller/BookController.java (hunk 1 above)
Comment (okuzan) [inline src/main/java/mate/academy/intro/controller/BookController.java]:
```suggestion
    public List<BookDto> getAll(@PageableDefault(page = 0, size = 10) Pageable pageable) {

```

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 3]
```diff
@@ -18,42 +21,50 @@
...
     public BookDto updateBookById(
             @PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto
     ) {
         return bookService.updateById(id, requestDto);
     }
 
     @GetMapping("/search")
+    @Operation(summary
+            = "Search for books", description = "Searches for books based on specific parameters")
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```
Comment (okuzan) [inline src/main/java/mate/academy/intro/controller/BookController.java]:
```suggestion
    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, @PageableDefault(page = 0, size = 10) Pageable pageable) {

```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**